pkg/networkserver/downlink.go

```diff
@@ -767,25 +836,22 @@ func (ns *NetworkServer) scheduleDownlinkByPaths(ctx context.Context, req *sched
 			},
 		}
 		queuedEvents = append(queuedEvents, req.AttemptEvent(ctx, req.EndDeviceIdentifiers, down))
-
 		logger.WithField("path_count", len(req.DownlinkPaths)).Debug("Schedule downlink")
-		cc, err := a.peer.Conn()
-		if err != nil {
-			queuedEvents = append(queuedEvents, req.FailEvent(ctx, req.EndDeviceIdentifiers, err))
-			errs = append(errs, err)
-			continue
-		}
-		res, err := ttnpb.NewNsGsClient(cc).ScheduleDownlink(ctx, down, ns.WithClusterAuth())
+		delay, err := a.Schedule(ctx, down, ns.WithClusterAuth())
 		if err != nil {
 			queuedEvents = append(queuedEvents, req.FailEvent(ctx, req.EndDeviceIdentifiers, err))
 			errs = append(errs, err)
 			continue
 		}
-		logger.WithField("transmit_delay", res.Delay).Debug("Scheduled downlink")
+		transmitAt := timeNow().Add(delay)
+		logger.WithFields(log.Fields(
+			"transmission_delay", delay,
+			"transmit_at", transmitAt,
+		)).Debug("Scheduled downlink")
 		return &scheduledDownlink{
 			Message:    down,
-			TransmitAt: timeNow().Add(res.Delay),
-		}, append(queuedEvents, req.SuccessEvent(ctx, req.EndDeviceIdentifiers, res)), nil
+			TransmitAt: transmitAt,
+		}, append(queuedEvents, req.SuccessEvent(ctx, req.EndDeviceIdentifiers, nil)), nil
```

Comment:
Successful GS scheduling still should trigger a `SuccessEvent` containing the response.

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/2225/commits/9e1b191e549295ed2c17450f142d64baa3f7afcf